src/zimscraperlib/download.py

```diff
@@ -7,17 +7,33 @@
 
 import requests
 
+from . import logger
+
 WGET_BINARY = os.getenv("WGET_BINARY", "/usr/bin/wget")
 
 
-def save_file(url, fpath):
-    """ download a binary file from its URL """
-    req = requests.get(url)
-    req.raise_for_status()
-    if not fpath.parent.exists():
-        fpath.parent.mkdir(exist_ok=True)
-    with open(fpath, "wb") as fp:
-        fp.write(req.content)
+def save_file(url, fpath, timeout=30, retries=0):
+    """ download a file from its URL, and return headers
+
+        Only recommended to be used with small files/HTMLs """
+
+    max_attempts = retries + 1
+    for attempt in range(max_attempts):
+        try:
+            resp = requests.get(url, timeout=timeout)
+            resp.raise_for_status()
+            if not fpath.parent.exists():
```

Comment:
I don't think this should be handled here. What if the parent's parent does not exists ? We should assume the user secures the destination folder I think.

Reply:
Okay, we can assume that, or just add another argument to mkdir, `parents=True`